connection.go

```diff
@@ -506,6 +515,8 @@ func (c *Connection) writeLoop() {
 	var err error
 
 	for err == nil {
+		idleTimeTimer := time.NewTimer(c.Opts.IdleTime)
```

Comment:
I think we can fo something like this


```suggestion
	idleTimeTimer := time.NewTimer(c.Opts.IdleTime)
	defer idleTimeTimer.Stop()
	for err == nil {
```

and then, inside the `for loop` reset the timer: `idleTimeTimer.Reset(c.Opts.IdleTime)`

Reply:
Maybe you can explicitly drain the chan as discribed into godoc:

 ```
	if !t.Stop() {
		<-t.C
	}
	t.Reset(d)
```

Reply:
II need to think about it :)

Reply:
Maybe, we also encapsulate drain with select. Because`t.C` can be already drained (if expired), w/o select it will be deadlock here:

```go
   if !t.Stop() {
   	select{
   	case <-t.C:
   	default:
   	}
   }
   t.Reset(d)
```

Reply:
or `resetTimer` helperfunc ?

```go
func ResetTimer(t *time.Timer, d time.Duration) {
	if !t.Stop() {
		select {
		case <-t.C:
		default:
		}
	}

	t.Reset(d)
}
```